types/pg/index.d.ts

```diff
@@ -79,12 +80,12 @@ export interface QueryResultBase {
     fields: FieldDef[];
 }
 
-export interface QueryResult extends QueryResultBase {
-    rows: any[];
+export interface QueryResult<R = any> extends QueryResultBase {
+    rows: R[];
 }
 
-export interface QueryArrayResult extends QueryResultBase {
-    rows: any[][];
+export interface QueryArrayResult<R = any> extends QueryResultBase {
```

Comment:
Wouldn't this need to follow an approach similar to that of `QueryConfig`, where a tuple type is expected as the generic type? Otherwise, I don't see the generic type (`R`) as being useful at all, as all you can express is a matrix full of numbers, or one full of strings, and so on, effectively not allowing to express resultsets with columns of different types.

Reply:
Good catch, that's definitely a mistake.

Would changing it to this make sense, I'm too tired to tell right now:
```ts
export interface QueryArrayResult<R extends any[] = any[]> extends QueryResultBase {
    rows: R[];
}
```

Reply:
It is also somewhat late on this part of the planet, but it looks legit :wink: